__tests__/end-to-end.js

```diff
@@ -75,6 +80,23 @@ const testResults = {}
 const defaultTestTimeout = 100000
 const defaultJobTimeout = 100000
 
+const testDependencies = {}
+
+/**
+ * Recursively calculates all dependent tests and sets them in the
+ * testDependencies global.
+ */
+function addRecursiveDependencies (testName, dependentTests) {
+  const existingAndNew = dependentTests.concat(testDependencies[testName] || [])
+  testDependencies[testName] = [...(new Set(existingAndNew))]
+  // add additional dependencies that the dependentTests depend on
+  dependentTests.forEach(dependentTest => {
+    if (testDependencies[dependentTest]) {
+      addRecursiveDependencies(testName, testDependencies[dependentTest])
+    }
+  })
+}
+
 function makeMakeTest (defaultDependentTests: Array<string> | string = []) {
```

Comment:
Could this guy (`makeMakeTest`) have a comment? Any time a word is repeated in a variable/function name my brain tends to turn off.